linode/manager
# refactor: [M3-6478] - Make `<Select />` type safe

## Description 📝

## Why ❓
- You may have not noticed, but the Select component had basically **zero type-safety** resulting in very bad developer experience and opens the possibility for bugs. It has been like this for **years** 😥
- Refer to the before and after

## Description 
- Does a refactor / improvement of the "Enhanced" Select component
- The goal isn't to make the component perfect (because it is super messed up), but the goal is to make it better 
- Putting it up as a draft to get e2e test feedback and feedback from team

## Impact 📷

| Before 👎🏼  | After 🎉 |
|--------|-------|
| ![Screenshot 2023-04-11 at 8 47 10 PM](https://user-images.githubusercontent.com/115251059/231320018-2719eaf6-b6a9-4357-84eb-471c29a05c64.jpg)      | ![Screenshot 2023-04-11 at 8 46 17 PM](https://user-images.githubusercontent.com/115251059/231320041-ab4d221c-ffae-4b8f-9715-4d1bec5756c6.jpg)     |

| Before 👎🏼  | After 🎉 |
|--------|-------|
| ![Screenshot 2023-04-11 at 9 01 57 PM](https://user-images.githubusercontent.com/115251059/231321072-dfc1f5d2-eae7-4599-9756-117a154f07c0.jpg)      | ![Screenshot 2023-04-11 at 9 03 07 PM](https://user-images.githubusercontent.com/115251059/231321056-6ef3dea5-902f-4938-970f-bd7762fcfe63.jpg)     |

### Before 👎🏼  
https://user-images.githubusercontent.com/115251059/231321727-f9d26d44-9214-4a99-a7d1-cb8c673eba50.mov

### After 🎉
https://user-images.githubusercontent.com/115251059/231321739-e72bc643-febb-434e-9b51-ecc55303fba7.mov

## How to test 🧪

- Test places in the app that use a "Enhanced" Select Component
  - I recommend looking at the diff and going to each of the `<Select />` components that were modified and testing the functionality.
- The e2e tests (once they are working) should also help check for bugs
